Doc/library/string.rst

```diff
@@ -754,9 +754,22 @@ attributes:
   be set in the subclass's class namespace).
 
 * *idpattern* -- This is the regular expression describing the pattern for
-  non-braced placeholders (the braces will be added automatically as
-  appropriate).  The default value is the regular expression
-  ``[_a-z][_a-z0-9]*``.
+  non-braced placeholders.  The default value is the regular expression
+  ``[_a-z][_a-z0-9]*``.  If this is given and *braceidpattern* is ``None``
+  this pattern will also apply to braced placeholders (the braces will be
+  added automatically).
```

Comment:
I think it is actually a bit confusing to say "the braces will automatically be added" here.  Even the code doesn't do that, it is literally the case that braceidpattern defaults to idpattern, and I think that's all you need to say.  Or, rather, since this is the idpattern description, you could say "Also used as braceidpattern if branceidpattern is None."

Reply:
Thanks!  I agree that we can just remove the parenthetical comment.